modules/http/src/main/java/org/mule/runtime/module/http/internal/listener/grizzly/GrizzlyHttpServer.java

```diff
@@ -36,23 +39,33 @@
   private final HttpListenerRegistry listenerRegistry;
   private TCPNIOServerConnection serverConnection;
   private Supplier<Scheduler> schedulerSource;
+  private Runnable schedulerDisposer;
   private Scheduler scheduler;
   private boolean stopped = true;
   private boolean stopping;
   private String ownerName;
 
   public GrizzlyHttpServer(ServerAddress serverAddress, TCPNIOTransport transport, HttpListenerRegistry listenerRegistry,
-                           Supplier<Scheduler> schedulerSource) {
+                           Supplier<Scheduler> schedulerSource, Runnable schedulerDisposer) {
     this.serverAddress = serverAddress;
     this.transport = transport;
     this.listenerRegistry = listenerRegistry;
     this.schedulerSource = schedulerSource;
+    this.schedulerDisposer = schedulerDisposer;
   }
 
   @Override
   public synchronized void start() throws IOException {
     this.scheduler = schedulerSource != null ? schedulerSource.get() : null;
     serverConnection = transport.bind(serverAddress.getIp(), serverAddress.getPort());
+    serverConnection.addCloseListener((CloseListener<Closeable, ICloseType>) (closeable, type) -> {
+      try {
+        scheduler.stop(5000, MILLISECONDS);
```

Comment:
The timeout for this should be related to the graceful shutdown of mule. Can you get that value from the mule context and use it here somehow?. 

If it's not possible then we should add a todo at least.